src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -934,6 +1116,205 @@ - (void)readAttributesWithEndpointID:(NSNumber * _Nullable)endpointID
+                   auto onFailureCb = [establishedOrFailed, queue, subscriptionEstablished, reportHandler](
+                                          const app::ConcreteClusterPath * clusterPath, const uint32_t aValueId, CHIP_ERROR error) {
+                       if (!(*establishedOrFailed)) {
+                           *establishedOrFailed = YES;
+                           if (subscriptionEstablished) {
+                               dispatch_async(queue, subscriptionEstablished);
+                           }
+                       }
+                       if (reportHandler) {
+                           dispatch_async(queue, ^{
+                               reportHandler(nil, [MTRError errorForCHIPErrorCode:error]);
+                           });
+                       }
+                   };
+
+                   auto onEstablishedCb = [establishedOrFailed, queue, subscriptionEstablished]() {
+                       if (*establishedOrFailed) {
+                           return;
+                       }
+                       *establishedOrFailed = YES;
+                       if (subscriptionEstablished) {
+                           dispatch_async(queue, subscriptionEstablished);
+                       }
+                   };
+
+                   MTRReadClientContainer * container = [[MTRReadClientContainer alloc] init];
+                   container.deviceID = self.nodeID;
+
+                   std::vector<app::AttributePathParams> attributePathParamsList;
+                   std::vector<app::EventPathParams> eventPathParamsList;
+
+                   if (attributePaths != nil) {
+                       for (uint8_t i = 0; i < [attributePaths count]; i++) {
+                           chip::EndpointId endpointId = kInvalidEndpointId;
+                           chip::ClusterId clusterId = kInvalidClusterId;
+                           chip::AttributeId attributeId = kInvalidAttributeId;
+
+                           if ([attributePaths[i] endpoint]) {
+                               endpointId = static_cast<chip::EndpointId>([[attributePaths[i] endpoint] unsignedShortValue]);
+                           }
+
+                           if ([attributePaths[i] cluster]) {
+                               clusterId = static_cast<chip::ClusterId>([[attributePaths[i] cluster] unsignedLongValue]);
+                           }
+
+                           if ([attributePaths[i] attribute]) {
+                               attributeId = static_cast<chip::AttributeId>([[attributePaths[i] attribute] unsignedLongValue]);
+                           }
+
+                           attributePathParamsList.push_back(app::AttributePathParams(endpointId, clusterId, attributeId));
+                       }
+                       container.pathParamsSize = attributePathParamsList.size();
+                       container.pathParams = attributePathParamsList.data();
```

Comment:
This does not make sense.  The container needs to keep the paths alive.  But now it's just pointing into a stack variable.  How is it supposed to keep that alive?

Probably what should happen here is that move to using `new []` and `delete []` to allocate and free these things, with nothrow and null checks and whatnot.

Reply:
It's hard to agree with your opinion.
Even if it is passed to the stack, it is changed from SendRequest to TLV Format, so there is no problem even if it is made into a stack.
 
(Of course, free should be deleted below.)

Reply:
I'm sorry. I checked this variable has to make `new` and `delete`.

Reply:
modified